sdks/java/extensions/arrow/src/main/java/org/apache/beam/sdk/extensions/arrow/ArrowConversion.java

```diff
@@ -0,0 +1,448 @@
+      }
+    }
+
+    // TODO: Consider using ByteBuddyUtils.TypeConversion for this
+    private static class ArrowValueConverterVisitor
+        implements ArrowType.ArrowTypeVisitor<Function<Object, Object>> {
+      @Override
+      public Function<Object, Object> visit(ArrowType.Null type) {
+        throw new IllegalArgumentException("Type \'" + type.toString() + "\' not supported.");
+      }
+
+      @Override
+      public Function<Object, Object> visit(ArrowType.Struct type) {
+        // TODO: code to create a row.
+        return null;
+      }
+
+      @Override
+      public Function<Object, Object> visit(ArrowType.List type) {
+        return null;
+      }
+
+      @Override
+      public Function<Object, Object> visit(ArrowType.FixedSizeList type) {
+        throw new IllegalArgumentException("Type \'" + type.toString() + "\' not supported.");
+      }
+
+      @Override
+      public Function<Object, Object> visit(ArrowType.Union type) {
+        throw new IllegalArgumentException("Type \'" + type.toString() + "\' not supported.");
+      }
+
+      @Override
+      public Function<Object, Object> visit(ArrowType.Map type) {
+        throw new IllegalArgumentException("Type \'" + type.toString() + "\' not supported.");
+      }
+
+      @Override
+      public Function<Object, Object> visit(ArrowType.Duration type) {
+        throw new IllegalArgumentException("Type \'" + type.toString() + "\' not supported.");
+      }
+
+      @Override
+      public Function<Object, Object> visit(ArrowType.Int type) {
+        return null;
+      }
+
+      @Override
+      public Function<Object, Object> visit(ArrowType.FloatingPoint type) {
+        return null;
+      }
+
+      @Override
+      public Function<Object, Object> visit(ArrowType.Utf8 type) {
+        return (Object text) -> ((Text) text).toString();
+      }
+
+      @Override
+      public Function<Object, Object> visit(ArrowType.Binary type) {
+        return null;
+      }
+
+      @Override
+      public Function<Object, Object> visit(ArrowType.FixedSizeBinary type) {
+        return null;
+      }
+
+      @Override
+      public Function<Object, Object> visit(ArrowType.Bool type) {
+        return null;
+      }
+
+      @Override
+      public Function<Object, Object> visit(ArrowType.Decimal type) {
+        throw new IllegalArgumentException("Type \'" + type.toString() + "\' not supported.");
+      }
+
+      @Override
+      public Function<Object, Object> visit(ArrowType.Date type) {
+        throw new IllegalArgumentException("Type \'" + type.toString() + "\' not supported.");
+      }
+
+      @Override
+      public Function<Object, Object> visit(ArrowType.Time type) {
+        throw new IllegalArgumentException("Type \'" + type.toString() + "\' not supported.");
+      }
+
+      @Override
+      public Function<Object, Object> visit(ArrowType.Timestamp type) {
+        DateTimeZone tz = DateTimeZone.forID(type.getTimezone());
```

Comment:
is it ok for tz to be null or empty string here?  I forget the details on the java implementation but the Arrow spec differentiates between timestamps [with/without timezones](https://github.com/apache/arrow/blob/master/format/Schema.fbs#L178)

Reply:
Yes we should clarify this. I think joda DateTime always has a timezone, we may need to reject timezone naive timestamps.

Here's a stable reference to arrow spec regarding timezone-naive timestamps: https://github.com/apache/arrow/blob/6e29200cebeb94f6014c72d25b1bc3a1be9cff1c/format/Schema.fbs#L235-L239